third_party/move/move-model/bytecode/abstract_domain_derive/tests/test.rs

```diff
@@ -0,0 +1,80 @@
+use abstract_domain_derive::AbstractDomain;
+use move_stackless_bytecode::dataflow_domains::{AbstractDomain, JoinResult};
+
+#[derive(AbstractDomain)]
+struct Unit;
+
+/// Add a formal top and bottom element to type `T`
+#[derive(PartialEq, Eq, Clone)]
+pub enum Plus2<T> {
```

Comment:
I suggest renaming this as `FlatDomain<T>` since that's what it is (as the elements of `T` are considered all incomparable).  `Plus2` is a really cryptic name.

Reply:
`FlatDomain<T>` makes me think `T` is already a flat domain. It's called `Plus2` for now because people sometimes use the notation $2 = \\{ \top, \bot \\}$, and write $A + B$ as the disjoint union of $A$ and $B$. So `Plus2<T>` is $T + 2$ under this notation.